### PR TITLE
Added leftIfNull to Either.

### PR DESCRIPTION
**Pull request by atomcat1978**

### CODE REVIEW

File: modules/core/arrow-core/src/main/kotlin/arrow/core/Either.kt [hunk 1]
```diff
@@ -276,6 +276,23 @@ inline fun <A, B> EitherOf<A, B>.getOrHandle(crossinline default: (A) -> B): B =
...
+ * Example:
+ * ```
+ * Right(12).leftIfNull({ -1 }) // Result: Right(12)
+ * Right(null).leftIfNull({ -1 })  // Result: Left(-1)
+ *
+ * Left(12).leftIfNull({ -1 })      // Result: Left(12)
+ * ```
+ */
+inline fun <A, B> EitherOf<A, B?>.leftIfNull(crossinline default: () -> A): Either<A, B> =
+        fix().fold({ Left(it) }, { if ( it == null) Left(default()) else Right(it) })
```
[the remark this answers is not in the document]
Comment (atomcat1978) [reply modules/core/arrow-core/src/main/kotlin/arrow/core/Either.kt]: That's true.

File: modules/core/arrow-core/src/main/kotlin/arrow/core/Either.kt [hunk 2]
```diff
@@ -276,6 +276,23 @@ inline fun <A, B> EitherOf<A, B>.getOrHandle(crossinline default: (A) -> B): B =
...
+ * Example:
+ * ```
+ * Right(12).leftIfNull({ -1 }) // Result: Right(12)
+ * Right(null).leftIfNull({ -1 })  // Result: Left(-1)
+ *
+ * Left(12).leftIfNull({ -1 })      // Result: Left(12)
+ * ```
+ */
+inline fun <A, B> EitherOf<A, B?>.leftIfNull(crossinline default: () -> A): Either<A, B> =
+        fix().flatMap { if ( it == null) Left(default()) else Right(it) }
```
Comment (pakoito) [inline modules/core/arrow-core/src/main/kotlin/arrow/core/Either.kt]: I'm surprised our linter doesn't complain about the extra whitespace here

Comment (atomcat1978) [reply modules/core/arrow-core/src/main/kotlin/arrow/core/Either.kt]: U mean at the beginning. Yes, IntelliJ has added some extra here.

Comment (atomcat1978) [reply modules/core/arrow-core/src/main/kotlin/arrow/core/Either.kt]: Fixed. Before ```it == null``` as well.

**Comment by pakoito** [comment]
Aaaaand merged. Thanks @atomcat1978 !!

**Pull request merged**